File: apps/api/internal/models/team_member.go
package models

import (
	"time"

	"gorm.io/gorm"

	"megagig-software-solution/apps/api/internal/ids"
)

// TeamMember is a person shown on /team (and a Blog author). PhotoURL is a
// plain URL/path — an uploaded file or a static asset — rather than an
// Upload relation, so photos can be seeded from apps/web/public and replaced
// through the admin's image upload zone. Every social link is optional; the
// public site renders a button only for the ones that are set.
type TeamMember struct {
	ID          string         `gorm:"primarykey;size:36" json:"id"`
	Name        string         `gorm:"size:255" json:"name" binding:"required"`
	Role        string         `gorm:"size:255" json:"role" binding:"required"`
	PhotoURL    string         `gorm:"size:500" json:"photo_url"`
	LinkedinURL string         `gorm:"size:500" json:"linkedin_url"`
	GithubURL   string         `gorm:"size:500" json:"github_url"`
	TwitterURL  string         `gorm:"size:500" json:"twitter_url"`
	Published   bool           `json:"published"`
	SortOrder   int            `json:"sort_order"`
	Version     int            `gorm:"not null;default:1" json:"version"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
}

// BeforeCreate generates a UUID before inserting.
func (m *TeamMember) BeforeCreate(tx *gorm.DB) error {
	if m.ID == "" {
		m.ID = ids.New()
	}
	return nil
}

// BeforeUpdate increments Version so offline clients can detect server-side updates.
func (m *TeamMember) BeforeUpdate(tx *gorm.DB) error {
	tx.Statement.SetColumn("version", gorm.Expr("version + 1"))
	return nil
}

File: packages/shared/schemas/team-member.ts
import { z } from "zod";

// photo_url is a plain URL/path (uploaded file or static asset); every
// social link is optional — the public site shows a button only when set.
export const CreateTeamMemberSchema = z.object({
  name: z.string().min(1, "Required"),
  role: z.string().min(1, "Required"),
  photo_url: z.string().optional(),
  linkedin_url: z.string().optional(),
  github_url: z.string().optional(),
  twitter_url: z.string().optional(),
  published: z.boolean().optional(),
  sort_order: z.number().int().optional(),
});

export const UpdateTeamMemberSchema = z.object({
  name: z.string().min(1, "Required").optional(),
  role: z.string().min(1, "Required").optional(),
  photo_url: z.string().optional(),
  linkedin_url: z.string().optional(),
  github_url: z.string().optional(),
  twitter_url: z.string().optional(),
  published: z.boolean().optional(),
  sort_order: z.number().int().optional(),
});

export type CreateTeamMemberInput = z.infer<typeof CreateTeamMemberSchema>;
export type UpdateTeamMemberInput = z.infer<typeof UpdateTeamMemberSchema>;
